dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -296,37 +302,102 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
         co_await winrt::resume_background();
 
         std::wstring appId{ RetrieveNotificationAppId() };
+
         THROW_IF_FAILED(ToastNotifications_RemoveAllToastsForApp(appId.c_str()));
     }
 
+#if 1 //  Option 1
```

Comment:
First option, using scope_exit() to provide a single point for handling all telemetry.